pymeasure/instruments/redpitaya/redpitaya_scpi.py

```diff
@@ -0,0 +1,338 @@
+    (that should be activated on the board) and port is the TCP/IP port number, usually 5000
+
+    To activate the SCPI server, you have to connect first the redpitaya to your computer/network
+    and enter the url address written on the network plug (on the redpitaya). It should be something
+    like "RP-F06432.LOCAL/" then browse the menu, open the Development application and activate the
+    SCPI server. When activating the server, you'll be notified with the IP/port address to use
+    with this Instrument.
+    """
+
+    TRIGGER_SOURCES = ('DISABLED', 'NOW', 'CH1_PE', 'CH1_NE', 'CH2_PE', 'CH2_NE',
+                       'EXT_PE', 'EXT_NE', 'AWG_PE', 'AWG_NE')
+    LV_MAX = 1
+    HV_MAX = 20
+    CLOCK = 125e6  # Hz
+    DELAY_NS = tuple(np.array(np.array(range(-2**13, 2**13+1)) * 1 / CLOCK * 1e9, dtype=int))
+
+    def __init__(self,
+                 adapter=None,
+                 ip_address: str = '169.254.134.87', port: int = 5000, name="Redpitaya SCPI",
+                 read_termination='\r\n',
+                 write_termination='\r\n',
+                 **kwargs):
+
+        if adapter is None:  # if None build it from the usual way as written in the documentation
+            adapter = f"TCPIP::{ip_address}::{port}::SOCKET"
+
+        super().__init__(
+            adapter,
+            name,
+            read_termination=read_termination,
+            write_termination=write_termination,
+            **kwargs)
+
+    time = Instrument.control("SYST:TIME?",
+                              "SYST:TIME %s",
+                              """Control the time on board
+                              time should be given as a datetime.time object""",
+                              get_process=lambda _tstr:
+                              datetime.time(*[int(split) for split in _tstr]),
+                              set_process=lambda _time:
+                              _time.strftime('%H,%M,%S'),
+                              )
+
+    date = Instrument.control("SYST:DATE?",
+                              "SYST:DATE %s",
+                              """Control the date on board
+                              date should be given as a datetime.date object""",
+                              get_process=lambda dstr:
+                              datetime.date(*[int(split) for split in dstr]),
+                              set_process=lambda date: date.strftime('%Y,%m,%d'),
+                              )
+
+    board_name = Instrument.measurement("SYST:BRD:Name?",
+                                        """Get the RedPitaya board name""")
+
+    def digital_reset(self):
+        """Reset the state of all digital lines"""
+        self.write("DIG:RST")
+
+    dioN = Instrument.MultiChannelCreator(DigitalChannelN, list(range(7)), prefix='dioN')
+    dioP = Instrument.MultiChannelCreator(DigitalChannelP, list(range(7)), prefix='dioP')
+    led = Instrument.MultiChannelCreator(DigitalChannelLed, list(range(8)), prefix='led')
```

Comment:
It is better to put the channels close to the init, in order to see the instrument structure more easily.